stac-example/src/main/scala/geotrellis/stac/raster/StacItemOps.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.stac.raster
+
+import geotrellis.stac.extensions.proj.{ProjItemExtension, ProjTransform}
+import com.azavea.stac4s.StacItem
+import com.azavea.stac4s.extensions.eo.EOItemExtension
+import com.azavea.stac4s.syntax._
+import geotrellis.raster.{CellSize, Dimensions, GridExtent, RasterExtent}
+import geotrellis.proj4.{CRS, LatLng}
+import geotrellis.vector._
+import cats.syntax.apply._
+
+case class StacItemOps(self: StacItem) {
+  def eoExtension: Option[EOItemExtension]     = self.getExtensionFields[EOItemExtension].toOption
+  def projExtension: Option[ProjItemExtension] = self.getExtensionFields[ProjItemExtension].toOption
```

Comment:
similar to what we talked about earlier today in FP Guild, is there a reason to force to `Option` rather than leave it at the extension result until the last possible moment?

Reply:
@jisantuc eh, this one is a pretty dumb ops, that really does not trace missing extensions.

Reply:
I added them here https://github.com/geotrellis/geotrellis-server/pull/345/files#diff-8b287ff3b3ab2d2688318e42be341133f74850b0583ec6746e02e34e09c086bfR30-R34 but for all other methods I really only need `Option` (for now).